qcodes/instrument_drivers/QDev/QDac_channels.py

```diff
@@ -335,21 +337,30 @@ def _set_vrange(self, chan, switchint):
         parameter accordingly
         """
 
-        tdict = {'-10 V to 10 V': 0,
-                 '-1 V to 1 V': 1,
-                 10: 0,
-                 0: 0,
-                 1: 1}
+        self.write('vol {} {}'.format(chan, switchint))
 
-        old = tdict[self.channels[chan-1].vrange.get_latest()]
+        channel = self.channels[chan-1]
+        if channel.vrange.cache() != switchint:
+            cache = channel.v.cache
+            if switchint == 1:
+                cache(cache() * 0.1)
+            elif switchint == 0:
+                cache(cache() / 0.1)
```

Comment:
cache's `__call__` provides convenience only for `get()` call, not for `set(..)`
```suggestion
                cache.set(cache() / 0.1)
```

Reply:
Thank you for catching this. This will save me some time tomorrow testing with the qdac!